### ISSUE TITLE
What do you mean "environment"

### ISSUE THREAD

**Issue by ArchivingToolsForWBM** [mdn/content#19533]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/console/clear

### What specific section or headline is this issue about?

Text saying "if the environment allows it"

### What information was incorrect, unhelpful, or incomplete?

First of all, what do you mean "Environment"?

Second of all. does firefox have the option to disable executing ``console.clear()`` on the ``about:config``? This code does clear the console log and the browser console, despite the “persist logs” being checked.

### What did you expect to see?

A link to an explanation of how this "environment" works, and if there is an option to disable it.

**Comment by teoli2003** [comment]
Thanks, good catch!

I opened a PR to get this fixed. When it is merged, this issue will be automatically closed.

**Comment by ArchivingToolsForWBM** [comment]
Sites should not be able to forcibly clear the console log against the user's will, just like that JS cannot disable or affect the browser's toolbar, devtools, and many other features in any way. I could be using an extension, that prints out useful information. A browser is a user agent, not a site's agent.

I notice this: https://greasyfork.org/en/scripts/440060-anti-anti-debugger that someone made a greasemonkey script that counters any sites that uses quote on quote “DRM” to prevent the use of the console log via clearing it. While I have not seen such sites do this (that user didn't link to the site), I made my own HTML file with just a script:
```
<script>
	console.clear()
</script>
```
and discovered as explained earlier.

I've learned that extensions using obfuscated code is not allowed on firefox/chrome's extension store since it is primarily used  to evade security checks to submit malicious code.
